diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index e7f2597..30f65ce 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -81,6 +81,7 @@ ext.useExpoPublishing = {
   project.android {
     publishing {
       singleVariant("release") {
         withSourcesJar()
       }
     }
   }
 
   project.afterEvaluate {
+    if (project.components.findByName("release") != null) {
     publishing {
       publications {
         release(MavenPublication) {
@@ -93,6 +94,7 @@ ext.useExpoPublishing = {
       }
     }
+    }
   }
 }
 
